client/client.go

```diff
@@ -219,89 +219,97 @@ func (r *NotaryRepository) Initialize(rootKeyIDs []string, serverManagedRoles ..
+
+	if err := r.tufRepo.InitRoot(
+		rootRole,
+		timestampRole,
+		snapshotRole,
+		targetsRole,
+		false,
+	); err != nil {
+		logrus.Debug("Error on InitRoot: ", err.Error())
+		return err
+	}
+	if _, err := r.tufRepo.InitTargets(data.CanonicalTargetsRole); err != nil {
+		logrus.Debug("Error on InitTargets: ", err.Error())
+		return err
+	}
+	if err := r.tufRepo.InitSnapshot(); err != nil {
+		logrus.Debug("Error on InitSnapshot: ", err.Error())
+		return err
+	}
+
+	return r.saveMetadata(serverManagesSnapshot)
+}
+
+func (r *NotaryRepository) initializeRoles(rootKeys []data.PublicKey, localRoles, remoteRoles []string) (
+	root, targets, snapshot, timestamp data.BaseRole, err error) {
+
+	root = data.NewBaseRole(
+		data.CanonicalRootRole,
+		notary.MinThreshold,
+		rootKeys...,
 	)
 
 	// we want to create all the local keys first so we don't have to
 	// make unnecessary network calls
-	for _, role := range locallyManagedKeys {
+	for _, role := range localRoles {
 		// This is currently hardcoding the keys to ECDSA.
-		key, err := r.CryptoService.Create(role, r.gun, data.ECDSAKey)
+		var key data.PublicKey
+		key, err = r.CryptoService.Create(role, r.gun, data.ECDSAKey)
 		if err != nil {
-			return err
+			return
 		}
 		switch role {
 		case data.CanonicalSnapshotRole:
-			snapshotRole = data.NewBaseRole(
+			snapshot = data.NewBaseRole(
 				role,
 				notary.MinThreshold,
 				key,
 			)
 		case data.CanonicalTargetsRole:
-			targetsRole = data.NewBaseRole(
+			targets = data.NewBaseRole(
 				role,
 				notary.MinThreshold,
 				key,
 			)
 		}
 	}
-	for _, role := range remotelyManagedKeys {
+	for _, role := range remoteRoles {
 		// This key is generated by the remote server.
-		key, err := getRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
+		var key data.PublicKey
+		key, err = getRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
 		if err != nil {
-			return err
+			return
 		}
 		logrus.Debugf("got remote %s %s key with keyID: %s",
 			role, key.Algorithm(), key.ID())
 		switch role {
 		case data.CanonicalSnapshotRole:
-			snapshotRole = data.NewBaseRole(
+			snapshot = data.NewBaseRole(
 				role,
 				notary.MinThreshold,
 				key,
 			)
 		case data.CanonicalTimestampRole:
-			timestampRole = data.NewBaseRole(
+			timestamp = data.NewBaseRole(
 				role,
 				notary.MinThreshold,
 				key,
 			)
 		}
 	}
-
-	r.tufRepo = tuf.NewRepo(r.CryptoService)
-
-	err := r.tufRepo.InitRoot(
-		rootRole,
-		timestampRole,
-		snapshotRole,
-		targetsRole,
-		false,
-	)
-	if err != nil {
-		logrus.Debug("Error on InitRoot: ", err.Error())
-		return err
-	}
-	_, err = r.tufRepo.InitTargets(data.CanonicalTargetsRole)
-	if err != nil {
-		logrus.Debug("Error on InitTargets: ", err.Error())
-		return err
-	}
-	err = r.tufRepo.InitSnapshot()
-	if err != nil {
-		logrus.Debug("Error on InitSnapshot: ", err.Error())
-		return err
-	}
-
-	return r.saveMetadata(serverManagesSnapshot)
+	return root, targets, snapshot, timestamp, nil
```

Comment:
Although, I guess we will have already returned in both those cases if the err wasn't nil, but still a little implicit rather than explicit for my liking.